app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,633 @@
+                    menu:menu.clone(),
+                }
+            });
+
+        }
+        (layout, init)
+    }
+}
+
+/// Enum to indicate whether sections headers should be placed above or below a section. Dead code
+/// is allowed as only one option is used at compile time.
+#[allow(dead_code)]
+#[derive(Clone, Copy, Debug)]
+enum SectionHeaderPlacement {
+    Top,
+    Bottom,
+}
+
+/// Indicates whether sections headers should be placed above or below a section.
+const SECTION_HEADER_PLACEMENT: SectionHeaderPlacement = SectionHeaderPlacement::Bottom;
+
+
+// === Shape Definition ===
+
+mod background {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        (style:Style,bg_color:Vector4) {
+            let theme_path: style::Path = list_panel_theme::HERE.into();
+
+            let alpha = Var::<f32>::from(format!("({0}.w)",bg_color));
+            let bg_color = &Var::<color::Rgba>::from(bg_color.clone());
+
+            let content_width = style.get_number(theme_path.sub("content_width"));
+            let content_height = style.get_number(theme_path.sub("content_height"));
+            let content_corner_radius = style.get_number(theme_path.sub("content_corner_radius"));
+            let menu_divider_color = style.get_color(theme_path.sub("menu_divider_color"));
+            let menu_divider_height = style.get_number(theme_path.sub("menu_divider_height"));
+            let menu_height = style.get_number(theme_path.sub("menu_height"));
+
+            let width = content_width;
+            let height = content_height + menu_height;
+
+            let divider_y_pos = height / 2.0 - menu_height;
+
+            let left_width = &(width/2.0).px();
+            let left = Rect((left_width,height.px())).translate_x(-left_width/2.0);
+
+            let right_width = &(width/2.0 + 2.0 * content_corner_radius).px();
+            let right = Rect((right_width,height.px())).corners_radius(content_corner_radius.px());
+            let right = right.translate_x((width/4.0-content_corner_radius).px());
+
+            let divider = Rect((width.px(),menu_divider_height.px()));
+            let divider = divider.fill(menu_divider_color);
+            let divider = divider.translate_y(divider_y_pos.px());
+
+            let base_shape = &(left + right);
+            let background = base_shape.fill(bg_color);
+            let shadow     = shadow::from_shape_with_alpha(base_shape.into(),&alpha,style);
+
+            (shadow + background + divider).into()
+        }
+    }
+}
+
+mod hline {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        (style:Style) {
+            let theme_path: style::Path = list_panel_theme::HERE.into();
+
+            let width            = Var::<Pixels>::from("input_size.x");
+            let height           = Var::<Pixels>::from("input_size.y");
+
+            let section_divider_color = style.get_color(theme_path.sub("section_divider_color"));
+
```

Comment:
Again, I'm not sure why this spacing is done here. If you feeel its needed, maybe we should refactor some of the lines to another function? I sometimes get it - when defining a really complex shape (like an icon) - then we can use spacing between icon parts, but here, I dont see a reason really.